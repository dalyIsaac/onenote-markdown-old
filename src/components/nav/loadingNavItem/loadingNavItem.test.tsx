import * as enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import { Spinner, SpinnerSize } from "office-ui-fabric-react";
import * as React from "react";
import LoadingNavItem, { IPropsLoadingNavItem } from "./loadingNavItem";

enzyme.configure({ adapter: new Adapter() });

function setUp() {
  const props: IPropsLoadingNavItem = {
    type: "Generic item",
    value: 0
  };
  const wrapper = enzyme.mount(<LoadingNavItem {...props} />);
  return { props, wrapper };
}

describe("Components: LoadingNavItem", () => {
  test("It should render self and subcomponents", () => {
    const { wrapper, props } = setUp();

    const divLoadingItemWrapper = wrapper.find("div").first();
    expect(divLoadingItemWrapper.hasClass("loadingItemWrapper")).toBe(true);

    const spinner = wrapper.find(Spinner).first();
    expect(spinner.hasClass("loadingNavItemSpinner")).toBe(true);
    expect(spinner.props().size).toBe(SpinnerSize.xSmall);
    expect(spinner.parent().is("div.loadingItemWrapper")).toBe(true);

    const label = wrapper.find("label").first();
    expect(label.hasClass("loadingNavItemSpinnerLabel")).toBe(true);
    expect(label.parent().is("div.loadingItemWrapper")).toBe(true);
    expect(label.text()).toBe(`Loading 0 ${props.type}s`);
  });

  test("Ensures that the grammar of the label changes as the value updates", () => {
    const { wrapper, props } = setUp();

    const label = wrapper.find("label").first();
    expect(label.text()).toBe(`Loading 0 ${props.type}s`);

    wrapper.setProps({ value: 1 });
    expect(label.text()).toBe(`Loading 1 ${props.type}`);

    wrapper.setProps({ value: 2 });
    expect(label.text()).toBe(`Loading 2 ${props.type}s`);
  });
});
