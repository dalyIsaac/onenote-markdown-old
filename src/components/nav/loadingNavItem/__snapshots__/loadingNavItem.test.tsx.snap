// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: LoadingNavItem It should render self and subcomponents 1`] = `
<div
  className="loadingItemWrapper"
>
  <div
    className="ms-Spinner loadingNavItemSpinner root-42"
  >
    <div
      className="ms-Spinner-circle ms-Spinner--xSmall circle-43"
    />
  </div>
  <label
    className="loadingNavItemSpinnerLabel"
  >
    Loading 
    0
     
    Generic items
  </label>
</div>
`;
